crates/storage/src/lib.rs

```diff
@@ -62,9 +68,13 @@ mod test_utils;
 
 #[doc(inline)]
 pub use self::{
-    alloc::Box,
+    lazy::Mapping,
+    memory::Memory,
```

Comment:
Do we really need this? It feels like an anti pattern to make use of this. Data should be passed as an argument where possible IMHO. Also, I think supporting this in the future will be annoying.

Reply:
Tbh, I'm not sure. I haven't seen any `ink!` contracts using `Memory`, but I know it's used in Solidity contracts. Maybe @cmichi or @ascjones can chime in here?

Reply:
No strong opinion here. I have a bad feeling about not having any example using this though. In consequence we don't have any proper on-chain `ink-waterfall` test for it either.

>Data should be passed as an argument where possible IMHO.

Agree, but I don't know enough about why it was introduced in Solidity in the first place.

Does @ascjones know more?